src/ui/CategoryFacet/CategoryFacetSearch.ts

```diff
@@ -1,3 +1,199 @@
-// export class CategoryFacetSearch {
-// constructor() {}
-// }
+import { CategoryFacet } from './CategoryFacet';
+import { FacetSearchElement } from '../Facet/FacetSearchElement';
+import { debounce, last, first } from 'underscore';
+import { $$, Dom } from '../../utils/Dom';
+import { SVGDom } from '../../utils/SVGDom';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { l } from '../../strings/Strings';
+import { IGroupByValue } from '../../rest/GroupByValue';
+import { KEYBOARD } from '../../utils/KeyboardUtils';
+import 'styling/_CategoryFacetSearch';
+
+export class CategoryFacetSearch {
+  public container: Dom | undefined;
+  private facetSearchElement: FacetSearchElement;
+  private currentlyDisplayedResults: HTMLElement;
+  private displayNewValues: () => void;
+  private currentResult: Dom | undefined;
+
+  constructor(private categoryFacet: CategoryFacet) {
+    this.facetSearchElement = new FacetSearchElement();
+    this.displayNewValues = debounce(() => {
+      this.facetSearchElement.showFacetSearchWaitingAnimation();
+      this.categoryFacet.logger.info('Triggering new Category Facet search');
+      this.categoryFacet.categoryFacetQueryController
+        .searchFacetValues(this.facetSearchElement.input.value)
+        .then((categoryFacetValues: IGroupByValue[]) => {
+          if (categoryFacetValues.length == 0) {
+            this.noFacetSearchResults();
+            return;
+          }
+          this.withFacetSearchResults();
+          $$(this.facetSearchElement.searchResults).empty();
+          for (let i = 0; i < categoryFacetValues.length; i++) {
+            const searchResult = this.buildFacetSearchValue(categoryFacetValues[i]);
+            if (i == 0) {
+              this.setAsCurrentResult(searchResult);
+            }
+            this.facetSearchElement.searchResults.appendChild(searchResult.el);
+          }
+          this.facetSearchElement.positionSearchResults(
+            this.categoryFacet.root,
+            this.categoryFacet.element.clientWidth,
+            this.facetSearchElement.search
+          );
+          this.facetSearchElement.hideFacetSearchWaitingAnimation();
+        });
+    }, this.categoryFacet.options.facetSearchDelay);
+    this.categoryFacet.root.addEventListener('click', (e: MouseEvent) => this.handleClickElsewhere(e));
```

Comment:
Is there a place where the event listener is removed when the component is destroyed?

Reply:
I assumed that the component cannot be destroyed without the search interface being also destroyed. I could be wrong but I don't think dynamically removing a single component is something we need to handle.